src/doc/trpl/glossary.md

```diff
@@ -37,3 +37,69 @@ And `2 + (3 * 4)` would look like this:
    / \
   3   4
 ```
+
+### Algebraic data types
+
+Algebraic data type (ADT) is a mathematical type category which includes product
+types (tuples) and sum types (enums) (also called tagged unions).
+
+#### Product types
+
+Consider product types: tuples. In tuples, the type is defined by the subtypes
```

Comment:
Since we just said they're product types, I'd just go with "Consider tuples." Also, "With tuples" sounds better to me than 'In tuples' but I can't justify it well.